server/src/test/java/org/opensearch/ingest/IngestServiceTests.java

```diff
@@ -2609,6 +2689,117 @@ public void testResolvePipelines_whenUseTemplateV1() throws Exception {
+        when(mockSystemProcessorFactory.isSystemGenerated()).thenReturn(true);
+        when(mockSystemProcessor.isSystemGenerated()).thenReturn(true);
+
+        // We add a default pipeline to index metadata to verify we DO NOT resolve it (only system pipeline)
+        final IngestService ingestService = createIngestServiceWithProcessors();
+        final SystemIngestPipelineCache cache = ingestService.getSystemIngestPipelineCache();
+        final IndexMetadata indexMetadata = spy(
+            IndexMetadata.builder("idx")
+                .settings(settings(Version.CURRENT).put(IndexSettings.DEFAULT_PIPELINE.getKey(), "default-pipeline"))
+                .putMapping("{}")
+                .numberOfShards(1)
+                .numberOfReplicas(0)
+                .putAlias(AliasMetadata.builder("alias").writeIndex(true).build())
+                .build()
+        );
+
+        final Index index = new Index("idx", "uuid");
+        when(indexMetadata.getIndex()).thenReturn(index);
+        Metadata metadata = Metadata.builder().indices(Map.of("idx", indexMetadata)).build();
+
+        // First time create the pipeline and cache it
+        IndexRequest indexRequest = new IndexRequest("idx");
+        boolean hasPipeline = ingestService.resolveSystemIngestPipeline(indexRequest, indexRequest, metadata);
+        // verify
+        verifySystemPipelineResolvedSuccessfully("[idx/uuid]", hasPipeline, indexRequest, cache);
+        assertEquals("[idx/uuid]", indexRequest.getSystemIngestPipeline());
+        assertEquals(NOOP_PIPELINE_NAME, indexRequest.getPipeline());
+        assertEquals(NOOP_PIPELINE_NAME, indexRequest.getFinalPipeline());
+    }
+
+    public void testResolveSystemIngestPipeline_whenExistingIndexAndSystemPipelineDisabled_thenNoSystemPipeline() throws Exception {
+        final IngestService ingestService = createIngestServiceWithProcessors();
+        ingestService.getClusterService()
+            .getClusterSettings()
+            .applySettings(Settings.builder().put(IngestService.SYSTEM_INGEST_PIPELINE_ENABLED.getKey(), false).build());
+
+        final IndexMetadata indexMetadata = spy(
+            IndexMetadata.builder("idx")
+                .settings(settings(Version.CURRENT).put(IndexSettings.DEFAULT_PIPELINE.getKey(), "default-pipeline"))
+                .putMapping("{}")
+                .numberOfShards(1)
+                .numberOfReplicas(0)
+                .putAlias(AliasMetadata.builder("alias").writeIndex(true).build())
+                .build()
+        );
+        final Index index = new Index("idx", "uuid");
+        when(indexMetadata.getIndex()).thenReturn(index);
+        Metadata metadata = Metadata.builder().indices(Map.of("idx", indexMetadata)).build();
+
+        IndexRequest indexRequest = new IndexRequest("idx");
+        boolean hasPipeline = ingestService.resolveSystemIngestPipeline(indexRequest, indexRequest, metadata);
+
+        // verify
```

Comment:
We can verify using the same method `verifySystemPipelineResolvedSuccessfully` here as well

Reply:
This is the opposite case where we are verifying system pipeline is NOT being resolved, so it wouldn't make sense to verify with the same helper